overlord/hookstate/hookmgr.go

```diff
@@ -256,24 +268,30 @@ func (m *HookManager) doRunHook(task *state.Task, tomb *tomb.Tomb) error {
 		return err
 	}
 
-	if hookExists {
-		output, err := runHook(context, tomb)
-		if err != nil {
-			if hooksup.TrackError {
-				trackHookError(context, output, err)
-			}
-			err = osutil.OutputErr(output, err)
-			if hooksup.IgnoreError {
-				task.State().Lock()
-				task.Errorf("ignoring failure in hook %q: %v", hooksup.Hook, err)
-				task.State().Unlock()
-			} else {
-				if handlerErr := context.Handler().Error(err); handlerErr != nil {
-					return handlerErr
-				}
-
-				return fmt.Errorf("run hook %q: %v", hooksup.Hook, err)
+	// some hooks get hijacked, e.g. the core configuration
+	var output []byte
+	if f := m.hijacked(hooksup.Snap, hooksup.Hook); f != nil {
```

Comment:
we should find out if we have an "hijack" much earlier, so that this check:

        hookExists := info.Hooks[hooksup.Hook] != nil
        if !hookExists && !hooksup.Optional {
                return fmt.Errorf("snap %q has no %q hook", hooksup.Snap, hooksup.Hook)
        }

is not relevant in the case we have one. Otherwise we can hijack only optional hooks if their missing from the snap (which usually they will).